libs/pubsub/query/query_test.go

```diff
@@ -11,6 +12,47 @@ import (
 	"github.com/cometbft/cometbft/libs/pubsub/query"
 )
 
+func TestBigNumbers(t *testing.T) {
+	testCases := []struct {
+		s        string
+		events   map[string][]string
+		err      bool
+		matches  bool
+		matchErr bool
+	}{
+
+		{"account.balance <= 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000"}}, false, true, false},
+		{"account.balance <= 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000.0"}}, false, true, false},
+		{"account.balance <= 20000000000000000000", map[string][]string{"account.balance": {"10000000000000000000"}}, false, true, false},
+		{"account.balance <= 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000001"}}, false, false, false},
+		{"account.balance <= 10000000000000000002", map[string][]string{"account.balance": {"10000000000000000000.6"}}, false, true, false},
+		{"account.balance > 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000.6"}}, false, true, false},
```

Comment:
Does this also pass?
```
{"account.balance > 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000.1"}}, false, true, false},
```

Reply:
No it does not because the `10000000000000000000.1` will be converted to `10000000000000000000.1`.

Reply:
I will bring this up on Monday, it seems that the DB allows event values to be stored  as floats. But the indexer queries always get the event value and convert it an int. So you end up with this particular event not being detected. 
We can leave this behaviour for old versions but it feels incorrect. Converting to int might work for most cases but in this particular cases, we need to detect that this number is a float, then parse it and check that there is a decimal part if the non decimal part is the same. I think for main we should 100% do this. But I want us to reach consensus on whether we want to do this for 0.34/0.37

Reply:
From today's discussion, I understand that [my initial comment](https://github.com/cometbft/cometbft/pull/771#discussion_r1180588069) does not apply to `v0.37.x`/`v0.34.x` (which this PR is targeting), isn't it?

Reply:
Reverting this commit means that if there is a float in the DB, it will be converted to its value in int64.